lib/common/xml.c

```diff
@@ -1809,45 +1816,47 @@ find_xml_node(xmlNode * root, const char *search_path, gboolean must_find)
     return NULL;
 }
 
-/* As the name suggests, the perfect match is required for both node
-   name and fully specified attribute, otherwise, when attribute not
-   specified, the outcome is the first node matching on the name. */
-static xmlNode *
-find_entity_by_attr_or_just_name(xmlNode *parent, const char *node_name,
-                                 const char *attr_n, const char *attr_v)
-{
-    xmlNode *child;
+#define attr_matches(c, n, v) pcmk__str_eq(crm_element_value((c), (n)), \
+                                           (v), pcmk__str_none)
 
+/*!
+ * \internal
+ * \brief Find first XML child element matching given criteria
+ *
+ * \param[in] parent     XML element to search
+ * \param[in] node_name  If not NULL, only match children of this type
+ * \param[in] attr_n     If not NULL, only match children with an attribute
+ *                       of this name and a value of \p attr_v
+ * \param[in] attr_v     If \p attr_n and this are not NULL, only match children
+ *                       with an attribute named \p attr_n and this value
+ *
+ * \return Matching XML child element, or NULL if none found
+ */
+xmlNode *
```

Comment:
Is there a reason to use `xmlNode *` over `xmlNodePtr`, or the other way around?  We can't seem to decide.

Reply:
I think you're the only one who's used xmlNodePtr ;-)

I'm not a fan of type aliases that are longer than the original, but I don't have a strong preference either way. There's some argument to say that since libxml2 uses it in function signatures we should follow.

I do think we should eventually drop GListPtr, since that's our own definition and we're kind of stepping on glib's namespace there.

Reply:
I think I only started doing this because I was looking to do something with libxml and its documentation uses that type.  I'm not especially a fan of hiding that kind of information behind a typedef either.  I'd be happy to write or ACK a patch that gets rid of our usage of it.  I just wanted the same thing everywhere.

Reply:
I'm fine with going in either direction

Reply:
```
clumens@localhost:~/src/pacemaker$ grep -r "xmlNode \*" * | wc -l
2237
clumens@localhost:~/src/pacemaker$ grep -r xmlNodePtr * | wc -l
94
```

The choice seems fairly obvious.